File: Medium/String/3499. Maximize Active Section with Trade I.cpp
// 3499. Maximize Active Section with Trade I
class Solution {
public:
    int maxActiveSectionsAfterTrade(string s) {
        int n = s.size();

        int i = 0;

        int left_0 = 0;
        int total_0 = 0;
        int one = 0;


        while(i < n){
            if(s[i] == '0'){
                left_0++;
                i++;
                continue;
            }
            while(i<n && s[i] == '1'){
                one++;
                i++;
            }
            int right_0 = 0;
            while(i<n && s[i] == '0'){
                right_0++;
                i++;
            }
            if(left_0 >0 && right_0 >0) total_0 = max(total_0, left_0+right_0);
            left_0 = right_0;

        }
        return one+total_0;



        
    }
};

File: Medium/String/3499. Maximize Active Section with Trade I.py
# 3499. Maximize Active Section with Trade I
# app 2 (app 1 in c++)
class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        n = len(s)

        one = 0
        zero = []
        i = 0
        while(i<n):
            if(s[i] == '1'):
                one+=1
                i+=1
                continue
            temp = 0
            while(i<n and s[i] == '0'):
                temp+=1
                i+=1

            zero.append(temp)
        
        if(len(zero) == 1): return one

        ans = one
        for j in range(1,len(zero),1):
            ans = max(ans,one+zero[j]+zero[j-1])
        
        return ans
            
        
